File: lab3.py
#Task 1 (Hailstone Sequence (Collatz Conjecture)

n = int(input("Please enter a positive int greater than 1: "))

while n <= 1:
    n = int(input("Please enter a positive int greater than 1: "))

steps = 0
print(n, end="")

while n != 1:

    if n % 2 == 0:
        n = n // 2
    else:
        n = n * 3 + 1

    print(" ->", n, end="")
    steps += 1

print("\nTotal steps:", steps)
print("Now we are continuing with FızBuzz game. ")

#Task 2 (FizzBuzz Counter)
n = int(input("\nPlease enter a numb between 10 and 100: "))

while n < 10 or n > 100:
    print("Invalid input.")
    n = int(input("Please enter a number between 10 and 100: "))

fizz = 0
buzz = 0
fizzbuzz = 0

for i in range(1, n+1):

    if i % 7 == 0:
        print( i, " is skipped")

    if i % 3 == 0 and i % 5 == 0:
        print("FizzBuzz")
        fizzbuzz += 1

    elif i % 3 == 0:
        print("Fizz")
        fizz += 1

    elif i % 5 == 0:
        print("Buzz")
        buzz += 1

    else:
        print(i)


print("Fizz counter: " ,fizz)
print("Buzz counter:" ,buzz)
print("Cızbız counter :" ,fizzbuzz)

#Task 3

n = int(input("Please enter a number between 3 and 9: "))

for i in range(1, 2 * n):
    k = n - abs(n - i)
    
    for j in range(k):
        print("*", end="")

    print()

File: lab3.cpp
#include <iostream>
using namespace std;

int main() {
    // Task 1 (Hailstone Sequence (Collatz Conjecture)

    int n;
    cout << "Please enter a positive int greater than 1: ";
    cin >> n;

    while (n <= 1) {
        cout << "Please enter a positive int greater than 1: ";
        cin >> n;
    }

    int steps = 0;
    cout << n;

    while (n != 1) {

        if (n % 2 == 0) {
            n = n / 2;
        } else {
            n = n * 3 + 1;
        }

        cout << " -> " << n;
        steps++;
    }

    cout << "\nTotal steps: " << steps << endl;
    cout << "Now we are continuing with FizzBuzz game." << endl;

    // Task 2 (FizzBuzz Counter)

    cout << "\nPlease enter a number between 10 and 100: ";
    cin >> n;

    while (n < 10 || n > 100) {
        cout << "Invalid input." << endl;
        cout << "Please enter a number between 10 and 100: ";
        cin >> n;
    }
    int fizz = 0;
    int buzz = 0;
    int fizzbuzz = 0;

    for (int i = 1; i <= n; i++) {

        if (i % 7 == 0) {
            cout << i << " is skipped" << endl;
        }

        if (i % 3 == 0 && i % 5 == 0) {
            cout << "FizzBuzz" << endl;
            fizzbuzz++;
        }

        else if (i % 3 == 0) {
            cout << "Fizz" << endl;
            fizz++;
        }

        else if (i % 5 == 0) {
            cout << "Buzz" << endl;
            buzz++;
        }

        else {
            cout << i << endl;
        }
    }

    cout << "Fizz counter: " << fizz << endl;
    cout << "Buzz counter: " << buzz << endl;
    cout << "FizzBuzz counter: " << fizzbuzz << endl;


    //TaSK 3:
    int x;
    cout << "Please enter a number between 3 and 9: ";
    cin >> x;

    for (int i = 1; i < 2 * x; i++) {
        int k = x - abs(x - i);

        for (int j = 0; j < k; j++) {
            cout << "*";
        }

        cout << endl;
    }

    return 0;
}
